cmd/gcp/secret_manager.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"io"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+)
+
+// ECDSAWithSHA256Signer implements crypto.Signer using Google Cloud Secret Manager.
+// Only crypto.SHA256 and ECDSA are supported.
+type ECDSAWithSHA256Signer struct {
+	publicKey  crypto.PublicKey
+	privateKey crypto.PrivateKey
```

Comment:
These members can be `ecdsa.{Public,Private}Key` now, which means that you can move the type check on L57-60 into `NewSecretManagerSigner()` so that there will be an error at construction time, rather than only detecting the mis-configuration later when the first attempt to sign something happens.

Reply:
Done. `x509.ParseECPrivateKey` returns `*ecdsa.PrivateKey`, so the explicit type check is only done for public key.